### DISCUSSION TITLE
How to integrate AppCenter (or any simple crash reporting system)?

### DISCUSSION THREAD

**Discussion by alessandrofrancesconi** [AvaloniaUI/Avalonia#5815] [Q&A] [answered]
My project is composed by mobile (Android/iOS) and desktop (PC/MacOS) versions. The mobile version is build with Xamarin, while the desktop version is build with Avalonia.
Crash reporting is provided by [AppCenter](https://appcenter.ms/). I'm very happy with it... but as of today it does not seem to support Avalonia, while it works good on Xamarin.
In 2019 (!!) I opened a [feature request](https://github.com/microsoft/appcenter-sdk-dotnet/issues/1083) to the AppCenter team, but I did not see any progress so far.

- Has anyone managed to integrate AppCenter into an Avalonia app? Are there some "hacky" solutions maybe?
- If answer is a big "NO", do you suggest other solutions that are still free & simple (AppCenter is perfect in this regard)... and compatible with all my platforms?

**Comment by maxkatz6** [comment]
AppCenter should not support Avalonia. AppCenter should just work with any .net app, even console app, and then it will work with Avalonia too. You might need to implement AppCenter backend by your hands though or look for alternatives. In overall it should be API with methods like "SendCrashReport", without depending on OS or .net target framework. I think google analytics sdk for .net should be just fine, I haven't used it though.

**Comment by alessandrofrancesconi** [comment, marked as answer]
I've came back to inform that I've solved it. AppCenter can be integrated via simple HTTP requests, by looking at:
- https://docs.microsoft.com/en-us/appcenter/diagnostics/using-the-diagnostics-api
- https://docs.microsoft.com/en-us/appcenter/diagnostics/upload-crashes
- Sample codes in mobile SDK (it already uses HTTP request internally, so you can copy them)
